### PR TITLE
MTDSA-11611: Add messages and metadata case classes for v3 calc retri…

### PR DESCRIPTION
**Pull request by BetaDraconis**
…eval.

### CODE REVIEW

File: test/v2/connectors/CrystallisationConnectorSpec.scala [hunk 1]
```diff
@@ -16,12 +16,13 @@
...
 
+import common.models.domain.DownstreamTaxYear
 import mocks.{MockAppConfig, MockHttpClient}
 import uk.gov.hmrc.http.HeaderCarrier
-import v2.models.domain.{DownstreamTaxYear, EmptyJsonBody, Nino}
+import v2.models.domain.{EmptyJsonBody, Nino}
 import v2.models.outcomes.ResponseWrapper
 import v2.models.request.crystallisation.CrystallisationRequest
-import v2.models.response.common.DownstreamUnit
+import v2.models.response.common.DesUnit
```
Comment (vfryer12) [inline test/v2/connectors/CrystallisationConnectorSpec.scala:25]: The DesUnit import in the V2 Spec endpoints is not working, it has not been updated to DesUnit in v2. I believe it's still DownstreamUnit

Comment (vfryer12) [reply test/v2/connectors/CrystallisationConnectorSpec.scala:25]: I think due to a shuffle around with the layout a couple of imports are no longer pointing to the correct locations

TaxRateSpec models response import needs to be updated to `import v3.models.response.common.TaxRate._`

Base connectorSpec you have `import v3.models.response.common.DownstreamResponse` and DownstreamResponse is not located here so it’s not recognising DownstreamResponse

File: resources/public/api/conf/3.0/application.raml [hunk 2]
```diff
@@ -39,7 +39,11 @@ documentation:
...
+      /{taxYear}:
         uriParameters:
           taxYear: !include pathParameters/taxYearMinimum2018.raml
-        post: !include endpoints/trigger.raml
\ No newline at end of file
+        post: !include endpoints/trigger.raml
+        /{calculationId}:
+          uriParameters:
+            calculationId: !include pathParameters/calculationId.raml
+          get: !include endpoints/retrieve.raml
```
Comment (Mbrookesy) [inline resources/public/api/conf/3.0/application.raml]: It seems the commit for the scaffolding RAML got included in this PR

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

File: app/v3/models/response/retrieveCalculation/messages/Messages.scala [hunk 3]
```diff
@@ -16,12 +16,16 @@
...
 
-import play.api.libs.json.{Json, OWrites, Reads}
+import play.api.libs.json.{Json, OFormat}
 
-case class Messages(field: String)
 
-object Messages {
-  implicit val reads: Reads[Messages] = Json.reads[Messages]
+case class Message(id: String, message: String)
+
```
Comment (Mbrookesy) [inline app/v3/models/response/retrieveCalculation/messages/Messages.scala:23]: The `message` field should be `text`

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

File: app/v3/models/response/common/CalculationType.scala [hunk 4]
```diff
@@ -16,15 +16,24 @@
...
 
-  implicit val format: Format[CalculationType] = Enums.format[CalculationType]
-}
+  implicit val writes: Writes[CalculationType] = Enums.writes[CalculationType]
+
+  implicit val reads: Reads[CalculationType] = Enums.readsUsing[CalculationType]{
+    case "inYear" => `inYear`
+    case "crystallisation" => `endOfYear`
+    case "biss" => `biss`
+    case "POA" => `POA`
```
[the remark this answers is not in the document]
Comment (BetaDraconis) [reply app/v3/models/response/common/CalculationType.scala:37]: I'm waiting to see what the results of our outstanding questions on this field are before making any changes.

**Pull request merged**